job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy

```diff
@@ -1359,6 +1359,8 @@ class PublisherContext extends AbstractExtensibleContext {
             disallowOwnCode(pullRequestPublisherContext.disallowOwnCode)
             onlyTriggerPhrase(pullRequestPublisherContext.onlyTriggerPhrase)
             mergeComment(pullRequestPublisherContext.mergeComment ?: '')
+            failOnNonMerge(pullRequestPublisherContext.failOnNonMerge)
```

Comment:
To maintain compatibility with older versions of the ghprb plugin, generate these elements only when a matching version is installed. See https://github.com/jenkinsci/job-dsl-plugin/blob/job-dsl-1.37/job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy#L316 for an example.